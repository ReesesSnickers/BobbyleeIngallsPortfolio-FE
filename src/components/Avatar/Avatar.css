.portfolio-avatar-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.portfolio-avatar {
  width: 400px;
  height: 400px;
  border-radius: 50%;
  overflow: hidden;
}
.portfolio-avatar-border {
  border: 1px solid #0e1a3c;
}
.portfolio-avatar-image {
  object-fit: fill;
  width: 100%;
  height: auto;
}
.portfolio-avatar-name {
  margin: 35px 0px 5px 0px;
  font-size: 48px;
  font-family: Helvetica, Arial, sans-serif;
  color: #0e1a3c;
  text-align: center;
}
.portfolio-avatar-title {
  margin: 0;
  font-size: 38px;
  font-family: Helvetica, Arial, sans-serif;
  color: #0e1a3c;
  text-align: center;
}
.portfolio-avatar-user {
  width: 400px;
}
