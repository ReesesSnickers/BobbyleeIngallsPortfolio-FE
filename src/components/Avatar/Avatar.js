import React from "react";
import PropTypes from "prop-types";

const Avatar = ({ src, alt, border, name, title }) => {
  return (
    <section className="portfolio-avatar-wrapper">
      <section
        className={`portfolio-avatar ${border && "portfolio-avatar-border"}`}
      >
        <img className="portfolio-avatar-image" src={src} alt={alt} />
      </section>
      <section className="portfolio-avatar-user">
        <h1 className="portfolio-avatar-name">{name}</h1>
        <h2 className="portfolio-avatar-title">{title}</h2>
      </section>
    </section>
  );
};

export default Avatar;

Avatar.propTypes = {
  // path to the avatar image
  src: PropTypes.string.isRequired,
  // alternate text for if the image cannot be found
  alt: PropTypes.string,
  // boolean for if there is a border around the image or not
  border: PropTypes.bool,
  // Avater name text
  name: PropTypes.string,
  // Avatar title text
  title: PropTypes.string
};

Avatar.defaultProps = {
  alt: "Profile Photo",
  border: false
};
