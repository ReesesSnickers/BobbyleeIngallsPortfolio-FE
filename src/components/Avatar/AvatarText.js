import React from "react";
import PropTypes from "prop-types";

const AvatarText = ({ primary, secondary, customClassName }) => {
  return (
    <section
      className={`wrapper${customClassName ? ` ${customClassName}` : ""}`}
    >
      <h1>{primary}</h1>
      <h2>{secondary}</h2>
      <style jsx>{`
        section.wrapper {
          width: 400px;
        }
        h1 {
          margin: 35px 0px 5px 0px;
          font-size: 48px;
          font-family: Helvetica, Arial, sans-serif;
          color: #0e1a3c;
          text-align: center;
        }
        h2 {
          margin: 0;
          font-size: 38px;
          font-family: Helvetica, Arial, sans-serif;
          color: #0e1a3c;
          text-align: center;
        }
        @media screen and (max-width: 930px) {
          h1 {
            font-size: 30px;
            margin-top: 20px;
          }
          h2 {
            font-size: 18px;
          }
        }

        @media screen and (max-width: 400px) {
          h1 {
            font-size: 25px;
          }
          h2 {
            font-size: 18px;
          }
        }
      `}</style>
    </section>
  );
};

export default AvatarText;

AvatarText.propTypes = {
  primary: PropTypes.string.isRequired,
  secondary: PropTypes.string.isRequired,
  customClassName: PropTypes.string,
};
