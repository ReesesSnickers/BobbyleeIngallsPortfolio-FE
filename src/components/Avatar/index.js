import "./Avatar.css";
export { default } from "./Avatar";
