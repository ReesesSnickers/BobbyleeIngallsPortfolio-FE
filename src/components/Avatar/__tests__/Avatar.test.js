import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import Avatar from "../Avatar.js";

let props;
let mountedComponent;

describe("Avatar", () => {
  const testComponent = () => {
    if (!mountedComponent) {
      mountedComponent = mount(<Avatar {...props} />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    props = {
      src: "MockSrc",
      alt: undefined,
      border: undefined,
      name: undefined,
      title: undefined
    };
    mountedComponent = undefined;
  });

  // initial render
  it("Renders the Avatar component", () => {
    let section = testComponent().find("section.portfolio-avatar-wrapper");
    expect(section.length).toBe(1);
  });

  // snapshot
  it("should match snapshot", () => {
    const tree = renderer.create(<Avatar {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // src
  it("should pass the src prop to the img src prop", () => {
    const image = testComponent().find("img");
    expect(image.props().src).toBe("MockSrc");
  });

  // alt default
  it("should pass the default alt prop to the img alt prop", () => {
    const image = testComponent().find("img");
    expect(image.props().alt).toBe("Profile Photo");
  });

  // border default
  it("should use the default border prop", () => {
    const section = testComponent()
      .find("section")
      .at(1);
    expect(section.hasClass("portfolio-avatar-border")).toBe(false);
  });

  // alt defined
  describe("When alt is defined", () => {
    beforeEach(() => {
      props.alt = "MockAlt";
    });
    it("should pass the alt prop to the img alt prop", () => {
      const image = testComponent().find("img");
      expect(image.props().alt).toBe("MockAlt");
    });
  });

  // border false
  describe("When border is defined false", () => {
    beforeEach(() => {
      props.border = false;
    });

    it("Will not use the border classname", () => {
      const section = testComponent()
        .find("section")
        .at(1);
      expect(section.hasClass("portfolio-avatar-border")).toBe(false);
    });
  });

  // border true
  describe("When border is defined true", () => {
    beforeEach(() => {
      props.border = true;
    });

    it("Will use the border classname", () => {
      const section = testComponent()
        .find("section")
        .at(1);
      expect(section.hasClass("portfolio-avatar-border")).toBe(true);
    });
  });

  // name defined
  describe("When name is defined", () => {
    beforeEach(() => {
      props.name = "MockName";
    });

    it("Will pass the name prop to the h1 inner text", () => {
      const header1 = testComponent().find("h1");
      expect(header1.text()).toBe("MockName");
    });
  });

  // title defined
  describe("When title is defined", () => {
    beforeEach(() => {
      props.title = "MockTitle";
    });

    it("Will pass the title prop to the h2 inner text", () => {
      const header2 = testComponent().find("h2");
      expect(header2.text()).toBe("MockTitle");
    });
  });
});
