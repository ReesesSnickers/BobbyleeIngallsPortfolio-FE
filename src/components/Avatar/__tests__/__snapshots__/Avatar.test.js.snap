// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should match snapshot 1`] = `
<section
  className="portfolio-avatar-wrapper"
>
  <section
    className="portfolio-avatar false"
  >
    <img
      alt="Profile Photo"
      className="portfolio-avatar-image"
      src="MockSrc"
    />
  </section>
  <section
    className="portfolio-avatar-user"
  >
    <h1
      className="portfolio-avatar-name"
    />
    <h2
      className="portfolio-avatar-title"
    />
  </section>
</section>
`;
